### DISCUSSION TITLE
Extension status

### DISCUSSION THREAD

**Discussion by nabajour** [gfx-rs/wgpu#3687] [Q&A]
Hi,

I made a small simulation program using compute using vulkan (with vulkano.rs). I'd like to port it to wgpu, to show it on my website if possible. 

My simulation uses some extensions, i64 and f64 math, and atomic operations on float. (in vulkano, it's the features:   shader_int64, shader_float64, shader_buffer_float32_atomics, shader_buffer_float32_atomic_add).

I got some of the operations working, (e.g. f64 math), but they seem only supported on Vulkan/native backend, using GLSL. 

If I understand correctly, I have to target the WebGPU backend, with WGSL, to be able to run my program in WebAssembly on my website, but these extensions don't seem to be part of the WebGPU standard.

Will those extensions be supported at some point ? It's difficult to do any interesting compute without this. 
If yes, is there something I can do in the meantime to be able to progress on my program? Like trying to get the Vulkan backend to work with the missing extensions? 
Is there some docs or examples I can follow to try to get it going?

Thanks

**Comment by nical** [comment]
> Will those extensions be supported at some point ?

Yes, right now browsers are busy shipping, polishing or catching up with what is shaping up to be the version 1 of the specification, but extensions like this are likely to be standardized next if they are supported by a large enough number of devices out there. It's an open standard with a couple of open source implementations so if you want to accelerate these things, a good way is to chime in, propose a spec and implementation, etc.
